kernel/packages/decentraland-ecs/src/ecs/Entity.ts

```diff
@@ -279,8 +280,19 @@ export class Entity implements IEntity {
   /**
    * Sets the parent entity
    */
-  setParent(_parent: IEntity | null): IEntity {
-    let newParent = !_parent && this.engine ? this.engine.rootEntity : _parent
+  setParent(_parent: IEntity | Attachable | null): IEntity {
+    let newParent: IEntity | null
+
+    // Check if parent is of type Attachable
+    if (_parent && 'getEntityRepresentation' in _parent) {
+      if (!this.engine) {
+        throw new Error(`In order to set an attachable as parent, you first need to add the entity to the engine.`)
+      }
+      newParent = _parent.getEntityRepresentation(this.engine)
+    } else {
+      // @ts-ignore
```

Comment:
```suggestion
```
I think this is not necessary.

Reply:
Weird. Locally with `make watch` it seems to work

Reply:
This is what failed
https://app.circleci.com/pipelines/github/decentraland/explorer/4918/workflows/d35fcba1-172d-4318-9d7a-fa97e81db84e/jobs/30707